ultravox/training/train.py

```diff
@@ -78,6 +80,19 @@ def main() -> None:
 
     transformers.set_seed(args.seed)
 
+    local_rank = int(os.environ.get("LOCAL_RANK", 0))
+    is_master = local_rank == 0
+
+    if args.do_train:
+        train(args)
+
+    if args.do_eval and is_master:
+        gc.collect()
+        torch.cuda.empty_cache()
+        evaluate(args)
+
+
+def train(args: config_base.TrainConfig):
     world_size = int(os.environ.get("WORLD_SIZE", 1))
     local_rank = int(os.environ.get("LOCAL_RANK", 0))
     is_master = local_rank == 0
```

Comment:
`is_master` is also computed earlier on line 84, maybe pass it in as an argument?

Reply:
It didn't seem like a big deal to keep it. Adding 3 new parameters to the function signature on the other hand didn't seem pretty.

Reply:
I see, all three variables are used in multiple places in `train`.